html2docx/image.py

```diff
@@ -30,7 +50,7 @@ def load_image(src: str) -> io.BytesIO:
                 # Read up to MAX_IMAGE_SIZE when response does not contain
                 # the Content-Length header. The extra byte avoids an extra read to
                 # check whether the EOF was reached.
-                data = response.read(MAX_IMAGE_SIZE + 1)
+                data = cast(bytes, response.read(MAX_IMAGE_SIZE + 1))
```

Comment:
If you're interested, we could contribute to [typeshed](https://github.com/python/typeshed/blob/master/stdlib/3/urllib/request.pyi) to avoid this type cast.

Reply:
I looked into this option. `urlopen()` returns either a `HTTPResponse` or an `addinfourl`. Both implement a `read()` that returns `bytes`. The return type used to be specified as the `Union` of both types, but [returning a `Union` type is discouraged](https://github.com/python/mypy/issues/1693) and the `urlopen()` was [changed to return type `Any`](https://github.com/python/typeshed/issues/3026).

An option might be to use structural subtyping and create a `Protocol`, so that methods common to `addinfourl` and `HTTPResponse` are hinted correctly. That would cover the `read()` method and probably a few others, but adds a maintenance burden.